homeassistant/components/greeneye_monitor/__init__.py

```diff
@@ -117,34 +116,32 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the GreenEye Monitor component."""
+    if server_config := config.get(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=server_config
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up greeneye_monitor from a config entry."""
     monitors = greeneye.Monitors()
     hass.data[DATA_GREENEYE_MONITOR] = monitors
 
-    server_config = config[DOMAIN]
-    await monitors.start_server(server_config[CONF_PORT])
+    port = config_entry.data[CONF_PORT]
+    await monitors.start_server(port)
 
     async def close_monitors(event: Event) -> None:
         """Close the Monitors object."""
         await monitors.close()
 
     hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, close_monitors)
 
-    num_sensors = 0
-    for monitor_config in config[DOMAIN][CONF_MONITORS]:
-        num_sensors += len(monitor_config[CONF_CHANNELS])
-        num_sensors += len(monitor_config[CONF_PULSE_COUNTERS])
-        num_sensors += len(monitor_config[CONF_TEMPERATURE_SENSORS][CONF_SENSORS])
-        num_sensors += len(monitor_config[CONF_VOLTAGE_SENSORS])
-
-    if num_sensors == 0:
-        _LOGGER.error(
-            "Configuration must specify at least one "
-            "channel, voltage, pulse counter or temperature sensor"
-        )
-        return False
-
     hass.async_create_task(
-        async_load_platform(hass, Platform.SENSOR, DOMAIN, config[DOMAIN], config)
+        hass.config_entries.async_forward_entry_setup(config_entry, "sensor")
```

Comment:
This should be adjusted to use: await async_forward_entry_setups() function instead.